packages/api/api.go

```diff
@@ -290,6 +291,10 @@ func NewServer(ctx context.Context, drv *sql.Driver, static *storage.BucketHandl
 	r.Handle("/playground", playground.Handler("GraphQL playground", "/query"))
 	r.Handle("/query", srv)
 
+	r.HandleFunc("/authentication/nonce", authentication.NonceHandler())
```

Comment:
i would prefer to expose this through our graphql endpoints, in which case we should do something similar to this: https://gqlgen.com/recipes/authentication/

Reply:
what you linked only refers to handling requests by having a middleware, are you talking about that or should we also implement the authentication endpoints under graphql? For now I've finished implementing a preprocessor for all of the upcoming requests and which sets wallet data to the context when a user is logged in